File: naive/484.swap-two-integers-in-array/Solution.java
import java.util.Arrays;

public class Solution{
    public void swapIntegers(int[] A, int index1, int index2){
        if(index1 < 0 || index2 < 0 || index1 >= A.length|| index2 >= A.length){
            return;
        }
        int tmp = A[index1];        
        A[index1] = A[index2];
        A[index2] = tmp;
    }
    public static void main(String[] args){
        Solution mySolution = new Solution();
        int[] a = {1,5,60,2};
        mySolution.swapIntegers(a,1,30);
        System.out.println(Arrays.toString(a));
    }
}

File: naive/484.swap-two-integers-in-array/Solution.go
package main

import (
	"fmt"
)

func main() {
	a := []int{1, 3, 5, 2}
	swapIntegers(&a, 1, -1)
	fmt.Println(a)
}

func swapIntegers(A *[]int, index1, index2 int) {
	p := *A
	if index1 >= len(p) || index2 >= len(p) || index1 < 0 || index2 < 0 {
		return
	}
	p[index1], p[index2] = p[index2], p[index1]
	A = &p
}
